### PR TITLE
Add the Capability to mass ban players based on .tsv files

### PR DESCRIPTION
**Pull request by Geekyhobo**
Yeah, it allows you to ban lots of people if you give it a tsv, gated behind the massban admin flag, give out this privilege with care

image

image

### CODE REVIEW

File: SS14.Admin/Pages/Bans/Create.cshtml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 ﻿@page
+@using Microsoft.AspNetCore.Mvc.TagHelpers
```
Comment (juliangiebel) [inline SS14.Admin/Pages/Bans/Create.cshtml:2]: Is that using actually required here? You didn't change anything else in this file

File: SS14.Admin/Pages/Bans/CreateMassBan.cshtml [hunk 2]
```diff
@@ -0,0 +1,91 @@
+﻿@page
+@model SS14.Admin.Pages.Bans.CreateMassBanModel
+
+@{
+    ViewData["Title"] = "Create New Mass Ban";
+}
+
+<partial name="Shared/StatusMessageAlert" />
```
Comment (juliangiebel) [inline SS14.Admin/Pages/Bans/CreateMassBan.cshtml:8]: Your indentation after this line is off

File: SS14.Admin/Pages/Bans/CreateMassBan.cshtml [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+            $('#confirmModal .close, #confirmModal button[data-dismiss="modal"]').click(function () {
+                $('#confirmModal').modal('hide'); // Close the modal
+            });
+
+            $('#confirmSubmit').click(function() {
+                $('#uploadForm').attr('action', '@Url.Page("CreateMassBan", "OnPostAsync")');
+            });
+        });
+    </script>
```
Comment (juliangiebel) [inline SS14.Admin/Pages/Bans/CreateMassBan.cshtml:90]: Couldn't you do the same you did for the ban confirmation PR you made?

File: SS14.Admin/Pages/Bans/CreateMassBan.cshtml [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+
+    <div class="container">
+        <h1>Mass Ban</h1>
+
+        <form method="post" enctype="multipart/form-data" id="uploadForm">
+            <div class="form-group">
+
+                <div class="custom-file m-1">
+                    <label for="file">Choose a .TSV file to Upload:</label>
```
Comment (juliangiebel) [inline SS14.Admin/Pages/Bans/CreateMassBan.cshtml:18]: You should probably explain the expected format somewhere. I.e. what columns and in which order and that you need a header row

File: SS14.Admin/Pages/Bans/CreateMassBan.cshtml.cs [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+            {
+                ModelState.AddModelError(string.Empty, "Please select a file.");
+                return Page();
+            }
+
+            if (!file.FileName.EndsWith(".tsv", StringComparison.OrdinalIgnoreCase))
+            {
+                ModelState.AddModelError(string.Empty, "Only TSV files are allowed.");
+                return Page();
+            }
```
Comment (juliangiebel) [inline SS14.Admin/Pages/Bans/CreateMassBan.cshtml.cs:47]: You might want to check what can cause an exception (missing header, wrong columns) and if some of that doesn't throw an exception but causes issues, for example that the first row gets ignored if there is no header, you should add a check for that here

File: SS14.Admin/Pages/Bans/Hits.cshtml [hunk 6]
```diff
@@ -1,6 +1,7 @@
 ﻿@page "{ban}"
 @using SS14.Admin.Helpers
 @using SS14.Admin.Pages.Tables
+@using Microsoft.AspNetCore.Mvc.TagHelpers
```
Comment (juliangiebel) [inline SS14.Admin/Pages/Bans/Hits.cshtml:4]: Is this needed?

File: SS14.Admin/Pages/Bans/Index.cshtml [hunk 7]
```diff
@@ -1,6 +1,7 @@
 ﻿@page
 @using SS14.Admin.Pages.Tables
 @using SS14.Admin.Helpers
+@using Microsoft.AspNetCore.Mvc.TagHelpers
```
Comment (juliangiebel) [inline SS14.Admin/Pages/Bans/Index.cshtml:4]: Same here

**Review by juliangiebel — CHANGES_REQUESTED** [review]

File: SS14.Admin/SS14.Admin.csproj [hunk 8]
```diff
@@ -10,6 +10,7 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="CsvHelper" />
```
Comment (PJB3005) [inline SS14.Admin/SS14.Admin.csproj:13]: You didn't add a matching entry to `Directory.Packages.props`?

File: SS14.Admin/Pages/Bans/CreateMassBan.cshtml.cs [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+        public CreateMassBanModel(PostgresServerDbContext dbContext, BanHelper banHelper, IWebHostEnvironment environment)
+        {
+            _dbContext = dbContext;
+            _banHelper = banHelper;
+            _environment = environment;
+        }
+
+        public int BanCount { get; private set; }
+
+        public class TsvEntry
```
Comment (PJB3005) [inline SS14.Admin/Pages/Bans/CreateMassBan.cshtml.cs:27]: Make this an actual C# `record` please.

**Review by PJB3005 — COMMENTED** [review]
Julian got most of the rest

File: SS14.Admin/Pages/Bans/Create.cshtml (hunk 1 above)
Comment (Geekyhobo) [reply SS14.Admin/Pages/Bans/Create.cshtml:2]: fixed

File: SS14.Admin/Pages/Bans/CreateMassBan.cshtml (hunk 2 above)
Comment (Geekyhobo) [reply SS14.Admin/Pages/Bans/CreateMassBan.cshtml:8]: fixed

File: SS14.Admin/Pages/Bans/CreateMassBan.cshtml (hunk 3 above)
Comment (Geekyhobo) [reply SS14.Admin/Pages/Bans/CreateMassBan.cshtml:90]: maybe, (yes i can)

File: SS14.Admin/Pages/Bans/Hits.cshtml (hunk 6 above)
Comment (Geekyhobo) [reply SS14.Admin/Pages/Bans/Hits.cshtml:4]: fixed

File: SS14.Admin/Pages/Bans/Index.cshtml (hunk 7 above)
Comment (Geekyhobo) [reply SS14.Admin/Pages/Bans/Index.cshtml:4]: fixed

File: SS14.Admin/Pages/Bans/CreateMassBan.cshtml (hunk 3 above)
Comment (Geekyhobo) [reply SS14.Admin/Pages/Bans/CreateMassBan.cshtml]: fixed

File: SS14.Admin/Pages/Bans/CreateMassBan.cshtml.cs (hunk 5 above)
Comment (Geekyhobo) [reply SS14.Admin/Pages/Bans/CreateMassBan.cshtml.cs:48]: its now checks

File: SS14.Admin/Pages/Bans/CreateMassBan.cshtml (hunk 4 above)
Comment (Geekyhobo) [reply SS14.Admin/Pages/Bans/CreateMassBan.cshtml]: done

File: SS14.Admin/SS14.Admin.csproj (hunk 8 above)
Comment (Geekyhobo) [reply SS14.Admin/SS14.Admin.csproj:13]: fixed

File: SS14.Admin/Pages/Bans/CreateMassBan.cshtml.cs (hunk 9 above)
Comment (Geekyhobo) [reply SS14.Admin/Pages/Bans/CreateMassBan.cshtml.cs]: done

**Comment by Geekyhobo** [comment]
Ready for re-reveiw, only a couple months later lol

**Review by juliangiebel — APPROVED** [review]
Looks good to me

**Pull request merged**